packages/fxa-auth-server/lib/routes/totp.js

```diff
@@ -201,42 +168,29 @@ module.exports = (log, db, mailer, customs, config) => {
         const sessionToken = request.auth.credentials;
         let exists = false;
 
-        return getTotpToken().then(() => {
-          return { exists };
-        });
+        if (sessionToken.tokenVerificationId) {
+          throw errors.unverifiedSession();
+        }
 
-        function getTotpToken() {
-          return P.resolve()
-            .then(() => {
-              if (sessionToken.tokenVerificationId) {
-                throw errors.unverifiedSession();
-              }
-
-              return db.totpToken(sessionToken.uid);
-            })
-
-            .then(
-              token => {
-                // If the token is not verified, lets delete it and report that
-                // it doesn't exist. This will help prevent some edge
-                // cases where the user started creating a token but never completed.
-                if (!token.verified) {
-                  return db.deleteTotpToken(sessionToken.uid).then(() => {
-                    exists = false;
-                  });
-                } else {
-                  exists = true;
-                }
-              },
-              err => {
-                if (err.errno === errors.ERRNO.TOTP_TOKEN_NOT_FOUND) {
-                  exists = false;
-                  return;
-                }
-                throw err;
-              }
-            );
+        try {
+          const token = await db.totpToken(sessionToken.uid);
+
+          // If the token is not verified, lets delete it and report that
+          // it doesn't exist. This will help prevent some edge
+          // cases where the user started creating a token but never completed.
+          if (!token.verified) {
+            await db.deleteTotpToken(sessionToken.uid);
+          }
```

Comment:
Lost the else statement here, exists should only be true if the token is verified.